File: src/main/java/tech/bobliu/assignment08/service/CaptchaService.java
package tech.bobliu.assignment08.service;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.InputStream;

public interface CaptchaService {
    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    class Captcha {
        private String filename;
        private String result;
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    class CaptchaResponse {
        private Captcha captcha;
        private int index;
    }

    CaptchaResponse getNextCaptcha(Integer lastIndex);
    InputStream getCaptchaImage(String filename);
}

File: src/main/kotlin/tech/bobliu/assignment08/service/CaptchaService.kt
package tech.bobliu.assignment08.service

import java.io.InputStream

interface CaptchaService {
    data class Captcha(val filename: String, val result: String)
    
    fun getNextCaptcha(lastIndex: Int?): Pair<Captcha, Int>
    fun getCaptchaImage(filename: String): InputStream
}
